xlsynth-g8r/src/equiv/prove_equiv.rs

```diff
@@ -742,9 +744,51 @@ pub fn ir_to_smt<'a, S: Solver>(
                     bitvec: bvs,
                 }
             }
-            NodePayload::Invoke { .. } => {
-                // TODO: add support for Invoke
-                panic!("Invoke not supported in SMT conversion");
+            NodePayload::Invoke { to_apply, operands } => {
+                let pkg = inputs
+                    .ir_fn
+                    .pkg_ref
+                    .expect("Invoke encountered but no package context available");
+                let callee = pkg
+                    .get_fn(to_apply)
+                    .unwrap_or_else(|| panic!("Invoke target '{}' not found in package", to_apply));
+
+                // Build callee inputs by mapping current env operands to callee params.
+                assert_eq!(
+                    callee.params.len(),
+                    operands.len(),
+                    "Invoke operand count does not match callee params"
+                );
+
+                let mut callee_input_map: HashMap<String, IrTypedBitVec<'a, S::Term>> =
+                    HashMap::new();
+                for (param, arg_ref) in callee.params.iter().zip(operands.iter()) {
+                    let arg_bv = env
+                        .get(arg_ref)
+                        .unwrap_or_else(|| panic!("Invoke arg BV missing for {}", param.name));
+                    callee_input_map.insert(
+                        param.name.clone(),
+                        IrTypedBitVec {
+                            ir_type: &param.ty,
+                            bitvec: arg_bv.bitvec.clone(),
+                        },
+                    );
+                }
+
+                let callee_ir_fn = IrFn {
+                    fn_ref: callee,
+                    pkg_ref: inputs.ir_fn.pkg_ref,
+                    fixed_implicit_activation: false,
+                };
+                let callee_inputs = FnInputs {
+                    ir_fn: &callee_ir_fn,
+                    inputs: callee_input_map,
+                };
+                let callee_smt = ir_to_smt(solver, &callee_inputs);
```

Comment:
I would maybe just make a comment to note this effectively inlines the callee in the SMT artifact